dspy/teleprompt/bootstrap_infer.py

```diff
@@ -0,0 +1,117 @@
+import copy
+import random
+import numpy as np
+from dspy.teleprompt import BootstrapFewShot
+from dspy.evaluate.evaluate import Evaluate
+import dspy
+    
+class BootstrapFewShotInfer(BootstrapFewShot):
```

Comment:
Can we add docstring to explain what this optimizer do? like what's the key improvement from BTFS, and when this optimizer should be used.